nebula_ros/include/nebula_ros/velodyne/velodyne_ros_wrapper.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "nebula_ros/common/mt_queue.hpp"
+#include "nebula_ros/common/parameter_descriptors.hpp"
+#include "nebula_ros/velodyne/decoder_wrapper.hpp"
+#include "nebula_ros/velodyne/hw_interface_wrapper.hpp"
+#include "nebula_ros/velodyne/hw_monitor_wrapper.hpp"
+
+#include <ament_index_cpp/get_package_prefix.hpp>
+#include <boost_tcp_driver/tcp_driver.hpp>
+#include <nebula_common/nebula_common.hpp>
+#include <nebula_common/nebula_status.hpp>
+#include <nebula_common/velodyne/velodyne_common.hpp>
+#include <nebula_hw_interfaces/nebula_hw_interfaces_velodyne/velodyne_hw_interface.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_components/register_node_macro.hpp>
+
+#include <nebula_msgs/msg/nebula_packet.hpp>
+#include <velodyne_msgs/msg/velodyne_scan.hpp>
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/asio.hpp>
+#include <boost/lexical_cast.hpp>
+
+#include <array>
+#include <chrono>
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <thread>
+#include <vector>
+
+namespace nebula
+{
+namespace ros
+{
+
+/// @brief Ros wrapper of velodyne driver
+class VelodyneRosWrapper final : public rclcpp::Node
+{
+public:
+  explicit VelodyneRosWrapper(const rclcpp::NodeOptions & options);
+  ~VelodyneRosWrapper() noexcept {};
+
+  /// @brief Get current status of this driver
+  /// @return Current status
+  Status GetStatus();
+
+  /// @brief Start point cloud streaming (Call CloudInterfaceStart of HwInterface)
+  /// @return Resulting status
+  Status StreamStart();
+
+private:
+  void ReceiveCloudPacketCallback(std::vector<uint8_t> & packet);
```

Comment:
nit: several method lack documentation

Reply:
I'll address that in a separate PR, a lot of the code needs documenting in other modules as well.